src/plugins/dbi_bam/src/SamtoolsBasedDbi.cpp

```diff
@@ -176,7 +176,7 @@ bamFile SamtoolsBasedDbi::openNewBamFileHandler() const {
     QString filePath = url.getURLString();
     NP<FILE> file = BAMUtils::openFile(filePath, "rb");
     bamFile bFile = bgzf_fdopen(file.getNullable(), "r");
-    CHECK_EXT(bFile != nullptr, BAMUtils::closeFileIfOpen(file), nullptr);
+    CHECK_EXT(bFile != nullptr, BAMUtils::closeFileIfOpen(file.getNullable()), nullptr);
```

Comment:
nit: it is not recommended to use getNullable at all, NP should be used like a regular raw pointer.
In this case, you do it like `bamFile bamHandler = file != nullptr ? bgzf_fdopen(file, "r"): nullptr;` and use `file` directly everywhere below without `getNullable`

Reply:
OK